src/CMakeLists.txt

```diff
@@ -527,9 +510,16 @@ set(libcommon_files
   common/fs_types.cc
   common/dns_resolve.cc
   common/hostname.cc
-  ${arch_files}
+  arch/probe.cc
   ${auth_files}
   ${mds_files})
+if(HAVE_ARM)
+  list(APPEND libcommon_files arch/arm.c)
+elseif(HAVE_INTEL)
+  list(APPEND libcommon_files arch/intel.c)
```

Comment:
There are symbols defined in src/arch/intel.c (e.g. ceph_arch_intel_pclmul and ceph_arch_intel_sse41) that are referenced in other parts of the code outside of any ifdef.  

I think the answer is probably to either (a) go back to unconditionally having arch/intel.c unconditionally (even on non-intel compiles) or (b) go forward with finding all references to those symbols and hiding them behind the proper intel related ifdefs.  

[100%] ?[32m?[1mLinking CXX executable ../../bin/radosgw-admin?[0m
[100%] Built target radosgw-admin
../../../lib/libcephd.a(libcephd.cc.o):(.toc+0x10): undefined reference to `ceph_arch_intel_pclmul'
../../../lib/libcephd.a(libcephd.cc.o):(.toc+0x18): undefined reference to `ceph_arch_intel_sse41'
collect2: error: ld returned 1 exit status
src/test/libcephd/CMakeFiles/ceph_test_cephd_api_misc.dir/build.make:147: recipe for target 'bin/ceph_test_cephd_api_misc' failed
make[2]: *** [bin/ceph_test_cephd_api_misc] Error 1
CMakeFiles/Makefile2:13507: recipe for target 'src/test/libcephd/CMakeFiles/ceph_test_cephd_api_misc.dir/all' failed
make[1]: *** [src/test/libcephd/CMakeFiles/ceph_test_cephd_api_misc.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[100%] ?[32m?[1mLinking CXX executable ../bin/ceph-dencoder?[0m
[100%] Built target ceph-dencoder

~/ceph/tchaikov.1/ceph/src/compressor/zlib$ cat CompressionPluginZlib.h | grep intel
#include "arch/intel.h"
      isal = (ceph_arch_intel_pclmul && ceph_arch_intel_sse41);
~/ceph/tchaikov.1/ceph/src/compressor/zlib$